lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -28,33 +35,48 @@ class JsUsage extends Gatherer {
   }
 
   /**
-   * @param {LH.Gatherer.PassContext} passContext
+   * @param {LH.Gatherer.FRTransitionalContext} context
    */
-  async beforePass(passContext) {
-    await passContext.driver.sendCommand('Profiler.enable');
-    await passContext.driver.sendCommand('Profiler.startPreciseCoverage', {detailed: false});
-
-    await passContext.driver.sendCommand('Debugger.enable');
-    await passContext.driver.on('Debugger.scriptParsed', this.onScriptParsed);
+  async startInstrumentation(context) {
+    const session = context.driver.defaultSession;
+    await session.sendCommand('Profiler.enable');
+    await session.sendCommand('Profiler.startPreciseCoverage', {detailed: false});
   }
 
   /**
-   * @param {LH.Gatherer.PassContext} passContext
-   * @return {Promise<LH.Artifacts['JsUsage']>}
+   * @param {LH.Gatherer.FRTransitionalContext} context
    */
-  async afterPass(passContext) {
-    const driver = passContext.driver;
+  async stopInstrumentation(context) {
+    const session = context.driver.defaultSession;
+    const coverageResponse = await session.sendCommand('Profiler.takePreciseCoverage');
+    this._scriptUsages = coverageResponse.result;
+    await session.sendCommand('Profiler.stopPreciseCoverage');
+    await session.sendCommand('Profiler.disable');
+  }
 
-    const coverageResponse = await driver.sendCommand('Profiler.takePreciseCoverage');
-    const scriptUsages = coverageResponse.result;
-    await driver.sendCommand('Profiler.stopPreciseCoverage');
-    await driver.sendCommand('Profiler.disable');
+  /**
+   * @param {LH.Gatherer.FRTransitionalContext} context
+   */
+  async startSensitiveInstrumentation(context) {
+    const session = context.driver.defaultSession;
+    await session.sendCommand('Debugger.enable');
+    await session.on('Debugger.scriptParsed', this.onScriptParsed);
+  }
 
-    await passContext.driver.sendCommand('Debugger.disable');
+  /**
+   * @param {LH.Gatherer.FRTransitionalContext} context
+   */
+  async stopSensitiveInstrumentation(context) {
+    await context.driver.defaultSession.sendCommand('Debugger.disable');
```

Comment:
we fire `Debugger.enable` by default so this won't actually disable the domain in legacy LH, probably more important to `.off` the `onScriptParsed`? this is probably source of some bugs today :)